module/purpose/Striker/src/Striker.cpp

```diff
@@ -0,0 +1,94 @@
+
+        on<Configuration>("Striker.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file Striker.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+        });
+
+        on<Provide<StrikerTask>, Optional<Trigger<GameState>>>().then(
+            [this](const StrikerTask& striker_task, const std::shared_ptr<const GameState>& game_state) {
+                // Do not use GameController information if force playing or force penalty shootout
+                if (striker_task.force_playing) {
+                    play();
+                    return;
+                }
+
+                // Check if there is GameState information, and if so act based on the current mode
+                if (game_state) {
+                    switch (game_state->data.mode.value) {
+                        case GameMode::PENALTY_SHOOTOUT: emit<Task>(std::make_unique<PenaltyShootoutStriker>()); break;
+                        case GameMode::NORMAL:
+                        case GameMode::OVERTIME: emit<Task>(std::make_unique<NormalStriker>()); break;
+                        default: log<NUClear::WARN>("Game mode unknown.");
+                    }
+                }
+            });
+
+        // Normal READY state
+        on<Provide<NormalStriker>, When<Phase, std::equal_to, Phase::READY>>().then(
+            [this] { emit<Task>(std::make_unique<Ready>()); });
+
+        // Normal PLAYING state
+        on<Provide<NormalStriker>, When<Phase, std::equal_to, Phase::PLAYING>>().then([this] { play(); });
+
+        // Normal UNKNOWN state
+        on<Provide<NormalStriker>, When<Phase, std::equal_to, Phase::UNKNOWN_PHASE>>().then(
+            [this] { log<NUClear::WARN>("Unknown normal game phase."); });
+
+        // Default for INITIAL, SET, FINISHED, TIMEOUT
+        on<Provide<NormalStriker>>().then([this] { emit<Task>(std::make_unique<StandStill>()); });
+
+        // Penalty shootout PLAYING state
+        on<Provide<PenaltyShootoutStriker>, When<Phase, std::equal_to, Phase::PLAYING>>().then([this] { play(); });
+
+        // Penalty shootout UNKNOWN state
+        on<Provide<PenaltyShootoutStriker>, When<Phase, std::equal_to, Phase::UNKNOWN_PHASE>>().then(
+            [this] { log<NUClear::WARN>("Unknown penalty shootout game phase."); });
+
+        // Default for INITIAL, READY, SET, FINISHED, TIMEOUT
+        on<Provide<PenaltyShootoutStriker>>().then([this] { emit<Task>(std::make_unique<StandStill>()); });
+    }
+
+    void Striker::play() {
+        // Walk to the ball and kick!
+        emit<Task>(std::make_unique<FindBall>(), 1);    // if the look/walk to ball tasks are not running, find the ball
+        emit<Task>(std::make_unique<LookAtBall>(), 2);  // try to track the ball
+        emit<Task>(std::make_unique<WalkToBall>(), 3);  // try to walk to the ball
+        emit<Task>(std::make_unique<KickTo>(Eigen::Vector3f::Zero()), 4);  // kick the ball if possible
```

Comment:
What's this syntax with the 1,2,3,4? A priority thing?

Reply:
yes priority

Reply:
I have added in a comment above these since priority is only really used in a few places higher up so ppl may not be super familiar.